components/suggest/src/rs.rs

```diff
@@ -69,17 +83,58 @@ pub(crate) trait Client {
     fn get_records(&self, request: RecordRequest) -> Result<Vec<Record>>;
 }
 
-impl Client for remote_settings::Client {
+/// Implements the [Client] trait using a real remote settings client
+pub struct RemoteSettingsClient {
+    // Create a separate client for each collection name
+    quicksuggest_client: remote_settings::Client,
+    #[cfg(feature = "fakespot")]
+    fakespot_client: remote_settings::Client,
+}
+
+impl RemoteSettingsClient {
+    pub fn new(
+        server: Option<remote_settings::RemoteSettingsServer>,
+        bucket_name: Option<String>,
+        server_url: Option<String>,
+    ) -> Result<Self> {
+        Ok(Self {
+            quicksuggest_client: remote_settings::Client::new(
+                remote_settings::RemoteSettingsConfig {
+                    server: server.clone(),
+                    bucket_name: bucket_name.clone(),
+                    collection_name: "quicksuggest".to_owned(),
+                    server_url: server_url.clone(),
+                },
+            )?,
+            #[cfg(feature = "fakespot")]
+            fakespot_client: remote_settings::Client::new(remote_settings::RemoteSettingsConfig {
+                server,
+                bucket_name,
+                collection_name: "fakespot-suggest-products".to_owned(),
+                server_url,
+            })?,
+        })
+    }
+}
+
+impl Client for RemoteSettingsClient {
     fn get_records(&self, request: RecordRequest) -> Result<Vec<Record>> {
+        let client = match request.record_type.as_str() {
```

Comment:
Or we can make this as a helper for RemoteSettingsClient, e.g. `RemoteSettingsClient.get_client(...)`?